### PR TITLE
electron app auto update performs a backup after download

### PR DESCRIPTION
**Pull request by jackkav**
Motivation:
In order to mitigate the risk of data loss.

Rationale:
We evaluated the trade offs of a full db backup and restore process against exporting each project. We found the restore process to be higher quality and simpler UX to use the existing export feature but trigger it before app updates.

Implementation notes:

- creates a folder in the current insomnia datapath: /backups/
- creates another version folder inside of backups: /2023.2.2/
- when an auto update is triggered and the download is complete, an export file will be created for every project in the database

Restore instructions:
- In the toolbar click Help then Show App Data folder, open backup folder and select the version folder and all project exports can be found inside and imported through the insomnia import feature.

Future work:

- add support for project import to the new import mechanism
- perhaps a backup cleaning process of some kind

Notes: 
The auto backup approach is triggered by the electron app auto upgrader functionality, which supports mac and windows, but not the portable binary on windows or linux, in those environments users can navigate to preferences -> data -> export -> export all, for each project they would like to backup.
Auto backup also incurs some necessary disk space usage, this is kept under control by including the version name in the folder, to ensure only the minimum number of necessary backups are made.

closes INS-2659

changelog(Improvements): Will export all projects to app data directory backups folder when auto update download completes

### CODE REVIEW

File: packages/insomnia/src/main.development.ts [hunk 1]
```diff
@@ -240,3 +244,15 @@ async function _trackStats() {
   });
   return stats;
 }
+async function backupDatabase() {
+  const dataPath = process.env['INSOMNIA_DATA_PATH'] || app.getPath('userData');
+  const files = await readdir(dataPath);
+  const backupPath = path.join(dataPath, 'backups', version);
+  await mkdir(backupPath, { recursive: true });
```
Comment (jackkav) [inline packages/insomnia/src/main.development.ts]: we could exit here if the backup folder exists, that way we would never overwrite a good db dbackup with a bad one.

**Review by filfreire — APPROVED** [review]
LGTM


We don't restore past responses, but I reckon this is far better than not being able to restore request data. I say  - ship it
image

**Comment by subnetmarco** [comment]
Backup/Restore should be a cloud capability.

**Comment by gatzjames** [comment]
Awesome 👏

**Pull request merged**